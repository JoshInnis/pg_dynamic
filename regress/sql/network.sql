/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */

--
-- Typecasting
--
SELECT ('192.168.1.5'::inet)::dynamic;
SELECT ('192.168.0.5'::inet)::dynamic;
SELECT ('192.168.1/24'::inet)::dynamic;
--

SELECT ('"192.168.1.5"::inet'::dynamic)::inet;
SELECT ('"192.168.0.5"::inet'::dynamic)::inet;
SELECT ('"192.168.1/24"::inet'::dynamic)::inet;


SELECT dynamic_toinet('"192.168.1/24"::inet');
SELECT dynamic_toinet('"192.168.1/24"');
SELECT dynamic_toinet('1');
SELECT dynamic_toinet('"1"');
SELECT dynamic_toinet('1.0');
SELECT dynamic_toinet('1::numeric');
SELECT dynamic_toinet('true');
SELECT dynamic_toinet('false');
SELECT dynamic_toinet('null');
SELECT dynamic_toinet('"2023-06-23 13:39:40.00"::timestamp');
SELECT dynamic_toinet('"1997-12-17 07:37:16-06"::timestamptz');
SELECT dynamic_toinet('"1997-12-17"::date');
SELECT dynamic_toinet('"07:37:16-08"::time');
SELECT dynamic_toinet('"07:37:16"::timetz');
SELECT dynamic_toinet('"30 Seconds"::interval');
SELECT dynamic_toinet('"192.168.1.5"::inet');
SELECT dynamic_toinet('"192.168.1.5"::cidr');
SELECT dynamic_toinet('[-1, 0, 1]');
SELECT dynamic_toinet('{"bool":true, "null":null, "string":"string", "integer":1, "float":1.2, "arrayi":[-1,0,1], "arrayf":[-1.0, 0.0, 1.0], "object":{"bool":true, "null":null, "string":"string", "int":1, "float":8.0}}');


--
-- inet && inet
--
SELECT '"192.168.1.5"::inet'::dynamic && '192.168.1/24::inet'::dynamic;
SELECT '"192.168.0.5"::inet'::dynamic && '"192.168.1/24"::inet'::dynamic;
SELECT '"192.168.1/24"::inet'::dynamic && '"192.168.1/24"::inet'::dynamic;