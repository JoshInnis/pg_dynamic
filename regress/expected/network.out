/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
--
-- Typecasting
--
SELECT ('192.168.1.5'::inet)::dynamic;
   dynamic   
-------------
 192.168.1.5
(1 row)

SELECT ('192.168.0.5'::inet)::dynamic;
   dynamic   
-------------
 192.168.0.5
(1 row)

SELECT ('192.168.1/24'::inet)::dynamic;
    dynamic     
----------------
 192.168.1.0/24
(1 row)

--
SELECT ('"192.168.1.5"::inet'::dynamic)::inet;
    inet     
-------------
 192.168.1.5
(1 row)

SELECT ('"192.168.0.5"::inet'::dynamic)::inet;
    inet     
-------------
 192.168.0.5
(1 row)

SELECT ('"192.168.1/24"::inet'::dynamic)::inet;
      inet      
----------------
 192.168.1.0/24
(1 row)

SELECT dynamic_toinet('"192.168.1/24"::inet');
 dynamic_toinet 
----------------
 192.168.1.0/24
(1 row)

SELECT dynamic_toinet('"192.168.1/24"');
 dynamic_toinet 
----------------
 192.168.1.0/24
(1 row)

SELECT dynamic_toinet('1');
ERROR:  cannot cast dynamic integer to type inet
SELECT dynamic_toinet('"1"');
ERROR:  invalid input syntax for type inet: "1"
SELECT dynamic_toinet('1.0');
ERROR:  cannot cast dynamic float to type inet
SELECT dynamic_toinet('1::numeric');
ERROR:  cannot cast dynamic numeric to type inet
SELECT dynamic_toinet('true');
ERROR:  cannot cast dynamic boolean to type inet
SELECT dynamic_toinet('false');
ERROR:  cannot cast dynamic boolean to type inet
SELECT dynamic_toinet('null');
ERROR:  cannot cast dynamic null to type inet
SELECT dynamic_toinet('"2023-06-23 13:39:40.00"::timestamp');
ERROR:  cannot cast dynamic timestamp to type inet
SELECT dynamic_toinet('"1997-12-17 07:37:16-06"::timestamptz');
ERROR:  cannot cast dynamic timestamptz to type inet
SELECT dynamic_toinet('"1997-12-17"::date');
ERROR:  cannot cast dynamic date to type inet
SELECT dynamic_toinet('"07:37:16-08"::time');
ERROR:  cannot cast dynamic time to type inet
SELECT dynamic_toinet('"07:37:16"::timetz');
ERROR:  cannot cast dynamic timetz to type inet
SELECT dynamic_toinet('"30 Seconds"::interval');
ERROR:  cannot cast dynamic interval to type inet
SELECT dynamic_toinet('"192.168.1.5"::inet');
 dynamic_toinet 
----------------
 192.168.1.5
(1 row)

SELECT dynamic_toinet('"192.168.1.5"::cidr');
ERROR:  unknown dynamic type: 13 can't cast to inet
SELECT dynamic_toinet('[-1, 0, 1]');
ERROR:  cannot cast non-scalar dynamic to dynamic inet
SELECT dynamic_toinet('{"bool":true, "null":null, "string":"string", "integer":1, "float":1.2, "arrayi":[-1,0,1], "arrayf":[-1.0, 0.0, 1.0], "object":{"bool":true, "null":null, "string":"string", "int":1, "float":8.0}}');
ERROR:  cannot cast non-scalar dynamic to dynamic inet
--
-- inet && inet
--
SELECT '"192.168.1.5"::inet'::dynamic && '192.168.1/24::inet'::dynamic;
ERROR:  operator does not exist: dynamic && dynamic
LINE 1: SELECT '"192.168.1.5"::inet'::dynamic && '192.168.1/24::inet...
                                              ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
SELECT '"192.168.0.5"::inet'::dynamic && '"192.168.1/24"::inet'::dynamic;
ERROR:  operator does not exist: dynamic && dynamic
LINE 1: SELECT '"192.168.0.5"::inet'::dynamic && '"192.168.1/24"::in...
                                              ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
SELECT '"192.168.1/24"::inet'::dynamic && '"192.168.1/24"::inet'::dynamic;
ERROR:  operator does not exist: dynamic && dynamic
LINE 1: SELECT '"192.168.1/24"::inet'::dynamic && '"192.168.1/24"::i...
                                               ^
HINT:  No operator matches the given name and argument types. You might need to add explicit type casts.
